### PR TITLE
Graduating GEP-726 (Path Redirects and Rewrites) to standard

### LINKED ISSUE #726: GEP: Add Path Redirects and Rewrites
**What would you like to be added**:
This GEP attempts to cover 2 related but different features:

* Path Redirects
* Rewrites

Once this GEP is complete, HTTPRoute should include filters that enable basic path and hostname redirects and rewrites. Although there is definitely variation among implementations as far as what is possible, there appears to be enough common ground for a portable API.

**Why is this needed**:
This is a common feature request, as noted by https://github.com/kubernetes-sigs/gateway-api/issues/200 and https://github.com/kubernetes-sigs/gateway-api/issues/678.

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This GEP has met the criteria for graduating to standard channel, as documented in https://github.com/kubernetes-sigs/gateway-api/issues/726#issuecomment-1419894406. This PR updates the GEP, documentation, and API types to reflect that.

**Which issue(s) this PR fixes**:
Fixes #726

**Does this PR introduce a user-facing change?**:
```release-note
GEP-726: Path Redirects and Rewrites has graduated to the standard channel.
```

/hold for consensus

### CODE REVIEW

**Review by Xunzhuo — COMMENTED** [review]
LGTM, thank!

**Review by youngnick — COMMENTED** [review]
/lgtm

/hold for multiple approval though.

File: apis/v1beta1/httproute_types.go
Comment (shaneutt) [inline apis/v1beta1/httproute_types.go:1]: Looks like we have a couple places like this:

```go
// <gateway:experimental>
type HTTPPathModifier struct {
```

Where the type itself is marked as experimental, do we want to change those now?

**Review by shaneutt — CHANGES_REQUESTED** [review]

**Comment by robscott** [comment]
Good catch, thanks @shaneutt! Missed that the first time but removed now. Removing hold since I think we've had enough approvals, next LGTM should merge this.

/hold cancel

File: apis/v1beta1/httproute_types.go
Comment (robscott) [reply apis/v1beta1/httproute_types.go:1]: Updated, thanks! (Only found one place, let me know if I missed something).

**Pull request merged**